### DISCUSSION TITLE
Moving `ghc-head` PRs to Gitlab

### DISCUSSION THREAD

**Discussion by Kleidukos** [haskell/haddock#1373] [Ideas]
@bgamari has suggested that it would be more convenient for everyone if we had the PRs destined to `ghc-head` opened and managed in the GitLab mirror. I am personally in favour of this, but I would like your opinion @hsyl20 @alexbiehl @sjakobi.

**Comment by bgamari** [comment]
To fill in a bit more context: Currently there are really two distinct types of development that Haddock sees:

 1. Functional changes to the documentation generator
 2. Changes simply updating `haddock-api` to accommodate changes in GHC

Under the proposed scheme, changes of type (2) would be managed on GitLab on the `ghc-head` branch whereas all other changes would remain here on GitHub. Periodically (e.g. ideally every month or two but at very least before every major GHC release) we would then merge the two branches.

**Comment by hsyl20** [comment]
Yes! It makes no sense to do the PR here where CI and reviewers can't check if it builds. In addition we could perhaps teach Marge to merge `wip/` branches to `ghc-head` => no more serialization of GHC's MRs.

**Reply by bgamari** [reply]
> Yes! It makes no sense to do the PR here where CI and reviewers can't check if it builds.

Agreed.

> In addition we could perhaps teach Marge to merge wip/ branches to ghc-head => no more serialization of GHC's MRs.

Yes, this would make life so much easier.

**Reply by simmsb** [reply]
Hey, I have CI functioning on the ghc-head branch here https://github.com/simmsb/haddock/actions/runs/698326272 
The required changes to haskell/actions are in this PR that I'm waiting on: https://github.com/haskell/actions/pull/42

**Reply by Kleidukos** [reply]
@simmsb sorry for the delay, I hope you'll be able to resume your work on the CI
